src/Renderer/ThreeExtended/FeaturesUtils.js

```diff
@@ -0,0 +1,122 @@
+function pointIsOverLine(point, linePoints, precision) {
+    const x0 = point._values[0];
+    const y0 = point._values[1];
+    // in first j is last point of line and then it'll be (i - 1)
+    for (var i = 0, j = linePoints.length - 1; i < linePoints.length; j = i++) {
+        const x1 = linePoints[i]._values[0];
+        const y1 = linePoints[i]._values[1];
+        const x2 = linePoints[j]._values[0];
+        const y2 = linePoints[j]._values[1];
+
+        const Xp = x0 - x1;
+        const Yp = y0 - y1;
+
+        const x21 = x2 - x1;
+        const y21 = y2 - y1;
+        const n = Math.sqrt(x21 * x21 + y21 * y21);
+        const scalar = (Xp * x21 + Yp * y21) / n;
+
+        if (scalar >= -precision && scalar <= n + precision) {
+            const distance = Math.abs(y21 * x0 - x21 * y0 + x2 * y1 - y2 * x1) / n;
+            if (distance <= precision) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+function getClosestPointIndice(point, points, precision) {
+    const x0 = point._values[0];
+    const y0 = point._values[1];
+    for (var i = 0; i < points.length; ++i) {
+        const x1 = points[i]._values[0];
+        const y1 = points[i]._values[1];
+        const xP = x0 - x1;
+        const yP = y0 - y1;
+        const n = Math.sqrt(xP * xP + yP * yP);
+        if (n < precision) {
+            return i;
```

Comment:
You're not returning the closest point.
You're returning the first point that matches `distance(point, points[i]) < precision`.

Reply:
(you can avoid a bunch of Math.sqrt as well by comparing against precisionSquared instead)

Reply:
> You're not returning the closest point.

I change that

> you can avoid a bunch of Math.sqrt as well by comparing against precisionSquared instead

yes, thanks